app/src/main/java/waiter/Connectable/ClientConnection.java

```diff
@@ -38,8 +40,36 @@ public void close() throws IOException {
         this.socket.close();
     }
 
-    private boolean notEndOfLine(StringBuilder requestStartLineBuilder) {
-        return requestStartLineBuilder.indexOf("\r\n") == -1;
+    private boolean notEndOfRequest(StringBuilder requestMessageBuilder) {
+        int requestBodyLength = 0;
+
+        if(headersWereRead(requestMessageBuilder)) {
+            requestBodyLength = getContentLength(requestMessageBuilder);
+            currentRequestBodyLength += 1;
+        }
+
+        return currentRequestBodyLength <= requestBodyLength;
+    }
+
+    private int getContentLength(StringBuilder requestMessageBuilder) {
+        int contentLength = 0;
+        String[] splitRequestMessage = requestMessageBuilder.toString().split("Content-Length: ");
+
+        if(contentLengthExists(splitRequestMessage)) {
+            contentLength = Integer.parseInt(
+                    splitRequestMessage[1].substring(0, splitRequestMessage[1].indexOf("\r\n"))
+            );
+        }
+
+        return contentLength;
+    }
+
+    private boolean contentLengthExists(String[] contentLengthHeader) {
+        return contentLengthHeader.length == 2;
```

Comment:
This is checking for a 2 because I use a `String.split()` to find the "Content-Length" header, which I then also use the result of here to check whether or not it split properly properly (if the length is two it found the Content-length or else it would be one). Looking at it more though, the name for the argument I pass in is a little misleading and the way it is checked for here is inconsistent with the other "request portion exists call". So I think Im going to check for the "Content-Length" using `indexOf` like the other one and combine the two function to be a generic `existsInRequestString` or something if that makes sense.